bql/semantic/hooks.go

```diff
@@ -664,6 +671,59 @@ func whereObjectClause() ElementHook {
 	return f
 }
 
+// whereFilterClause returns an element hook that updates the working filter clause and,
+// if the filter clause is complete, populates the filters list of the statement.
+func whereFilterClause() ElementHook {
+	var hook ElementHook
+	supportedFilterFunctions := map[string]bool{
+		"latest": true,
+	}
+
+	hook = func(st *Statement, ce ConsumedElement) (ElementHook, error) {
+		if ce.IsSymbol() {
+			return hook, nil
+		}
+
+		tkn := ce.Token()
+		currFilter := st.WorkingFilter()
```

Comment:
I am struggling to understand the scope of currFilter.
Is there any way to make it simpler?

Reply:
I have just refactored this filter hook to make it clearer.